engines/bastion_katello/lib/bastion_katello/engine.rb

```diff
@@ -33,6 +33,7 @@ class Engine < ::Rails::Engine
         ),
         :config => {
           'consumerCertRPM' => consumer_cert_rpm,
+          'enable_lazy_sync' => Setting[:enable_lazy_sync],
```

Comment:
I think this is the source of your test failure, as the engine is loaded before the Settings table exists.  You could maybe do:

`!!Foreman.in_rake? && Setting[:enable_lazy_sync]`

Make sure you're not in rake, and evaluate the setting.